### PR TITLE
[Pruning] Support of MultiHeadAttention layer for pruning.

### PR DESCRIPTION
**Pull request by wwwind**
Added support of MultiHeadAttention layer for pruning  + small model with this layer on mnist dataset: mnist_mha.py in examples directory.

### CODE REVIEW

File: tensorflow_model_optimization/python/examples/sparsity/keras/mnist/mnist_mha.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+pruning_params = {
+      'pruning_schedule': ConstantSparsity(0.75, begin_step=2000, frequency=100)
+}
+
+model_for_pruning = prune.prune_low_magnitude(model, **pruning_params)
+
+# `prune_low_magnitude` requires a recompile.
+model_for_pruning.compile(
+    optimizer="adam",
```
Comment (daverim) [inline tensorflow_model_optimization/python/examples/sparsity/keras/mnist/mnist_mha.py:83]: Should use single quotes every-where

File: tensorflow_model_optimization/python/examples/sparsity/keras/mnist/mnist_mha.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    train_labels,
+    batch_size=batch_size,
+    epochs=epochs,
+    callbacks=callbacks,
+    validation_split=validation_split,
+)
+
+score = model_for_pruning.evaluate(test_images, test_labels, verbose=0)
+print('Pruned model test loss:', score[0])
+print('Pruned model test accuracy:', score[1])
```
Comment (daverim) [inline tensorflow_model_optimization/python/examples/sparsity/keras/mnist/mnist_mha.py:99]: Add a new line?

**Review by daverim — APPROVED** [review]
Looks good to me

**Comment by wwwind** [comment]
Hi @daverim Thank you for the review. I addressed comments. Could you please re-approve ?

**Review by daverim — APPROVED** [review]

**Pull request merged**